lib/src/command/run_command.dart

```diff
@@ -46,8 +48,34 @@ class RunCommand extends MoronepoCommand<Null> {
           await stdout.addStream(process.stdout);
           await stderr.addStream(process.stderr);
           return process;
-        });
+        }).then((process) => process.exitCode);
+
+        if (code != 0) {
+          throw CommandException(code);
+        }
       });
+    } on CommandException catch (exception) {
+      exitCode = exception.exitCode;
     }
   }
+
+  Future<List<Project>> _findProjects(String rootDirectory, String projectName) async {
+    return ProjectFinder().find(
+      path: rootDirectory,
+      dependencies: moronepoResults.dependencies,
+      hasTests: moronepoResults.hasTests,
```

Comment:
What do you think about class `ProjectFilters`? It's a lot of parameters here and will be growing I think. It will be less occurrences if `MoronepoResults` return filters packed in object like this.